src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,1008 +50,333 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
+    }
+    
+    private void putAddrInMap(Address__c addr, Map<Id, Address__c> mapParentIdAddr) {
+        if(addr.Parent_Account__c != null) {
+            mapParentIdAddr.put(addr.Parent_Account__c, addr);
+        } else if(addr.Parent_Contact__c != null) {
+            mapParentIdAddr.put(addr.Parent_Contact__c, addr);
+        }
+    }
+    
+    private void processDml(DmlWrapper dmlWrapper, Map<Id, Address__c> mapParentIdAddr) {
         // we must process dmlWrapper ourselves, so we can extract out the appropriate errors
-        // that might occur updating contacts and accounts, and associating those errors with 
+        // that might occur updating contacts and parents, and associating those errors with 
         // the correct addresses.
         ERR_Handler.Errors errors = TDTM_TriggerHandler.processDMLWithRollback(dmlWrapper, false);
         if (errors.errorsExist) {
             for (Error__c error : errors.errorRecords) {
                 ID objId = ERR_Handler.getErrorObjectId(error);
                 UTIL_Debug.debug(LoggingLevel.WARN, '**** Address TDTM error on object: ' + objId);
-                if (UTIL_Describe.isObjectIdThisType(objId, 'Account')) {
-                    Address__c addr = mapAccIdAddr.get(objId);
-                    addr.addError(error.Full_Message__c);
-                    UTIL_Debug.debug(LoggingLevel.WARN, '**** Address: ' + addr + ' message: ' + error.Full_Message__c);
-                }
-                else if (UTIL_Describe.isObjectIdThisType(objId, UTIL_Namespace.StrTokenNSPrefix('Address__c'))) {
-                    Address__c addr = mapAddrIdAddr.get(objId);
+                if (UTIL_Describe.isObjectIdThisType(objId, 'Account') || UTIL_Describe.isObjectIdThisType(objId, 'Contact')) {
+                    Address__c addr = mapParentIdAddr.get(objId);
                     addr.addError(error.Full_Message__c);
                     UTIL_Debug.debug(LoggingLevel.WARN, '**** Address: ' + addr + ' message: ' + error.Full_Message__c);
                 }
             }
-            return null;
         }
-        
-        // hack to detect empty addresses and delete them after we've propagated the change to all contacts and account
+    }
+    
+    private void deleteEmptyAddrs(List<SObject> listNew, TDTM_Runnable.Action triggerAction) {
+        // hack to detect empty addresses and delete them after we've propagated the change to parent
         if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
-             List<Address__c> listAddr = new list<Address__c>();
+             List<Address__c> listAddr = new List<Address__c>();
              for (SObject so : listNew) {
                  Address__c addr = (Address__c)so;
-                 if (isAddressEmpty(addr))
+                 if (ADDR_Addresses_UTIL.isAddressEmpty(addr))
                     listAddr.add(new Address__c(Id = addr.Id));
              }
              if (listAddr.size() > 0)
                 delete listAddr; 
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.